sky/setup_files/setup.py

```diff
@@ -13,32 +13,91 @@
 discussion](https://github.com/skypilot-org/skypilot/discussions/1016)
 """
 
+import atexit
 import io
 import os
 import platform
 import re
+import subprocess
 from typing import Dict, List
-import warnings
 
 import setuptools
 
 ROOT_DIR = os.path.dirname(__file__)
+INIT_FILE_PATH = os.path.join(ROOT_DIR, 'sky', '__init__.py')
+original_init_content = None
 
 system = platform.system()
 
 
-def find_version(*filepath):
+def find_version():
     # Extract version information from filepath
     # Adapted from:
     #  https://github.com/ray-project/ray/blob/master/python/setup.py
-    with open(os.path.join(ROOT_DIR, *filepath)) as fp:
+    with open(INIT_FILE_PATH, 'r') as fp:
         version_match = re.search(r'^__version__ = [\'"]([^\'"]*)[\'"]',
                                   fp.read(), re.M)
         if version_match:
             return version_match.group(1)
         raise RuntimeError('Unable to find version string.')
 
 
+def get_commit_hash():
+    with open(INIT_FILE_PATH, 'r') as fp:
+        commit_match = re.search(r'^_SKYPILOT_COMMIT_SHA = [\'"]([^\'"]*)[\'"]',
+                                 fp.read(), re.M)
+        if commit_match:
+            commit_hash = commit_match.group(1)
+        else:
+            raise RuntimeError('Unable to find commit string.')
```

Comment:
Can we reuse the `find_version` function? Seems like a lot of code is duplicated

Reply:
It is short enough, maybe it is ok to duplicate a bit? Making it another function may take longer lines.

Reply:
Make sense. Agreed 👍